AssociateContactAccount.trigger

```diff
@@ -0,0 +1,15 @@
+/*
+ABC Containers require the ability to automatically associate a Contact created in their Salesforce Instance 
+with the respective Account based on the email domain specified in the primary email address of the Contact. 
+The association should happen real time as soon as a Contact record is created within the system.
+
+*/
+
+trigger AssociateContactAccount on Contact (before insert) {
+	AssociateContactAccountHelper helper = new AssociateContactAccountHelper();
```

Comment:
This object only suppose to be created only when it's needed. You can move this in if block.